lib/checks/navigation/heading-order-evaluate.js

```diff
@@ -3,6 +3,39 @@ import { querySelectorAllFilter, getAncestry } from '../../core/utils';
 import { isVisible } from '../../commons/dom';
 import { getRole } from '../../commons/aria';
 
+export default headingOrderEvaluate;
+
+function headingOrderEvaluate() {
+  // find all headings, even ones that are outside the current
+  // context. also need to know where iframes are so we can insert
+  // the results of any in-context iframes into their proper order
+  // @see https://github.com/dequelabs/axe-core/issues/728
+
+  const headingOrder = cache.get('headingOrder', getHeadingOrder());
+
+  this.data({ headingOrder });
+
+  return true;
```

Comment:
I would say this function doesn't benefit from the cache get default value pattern. The main reason is that the setting of `data` should happen inside `getHeadingOrder` and not in the evaluate function (we only want the data on the first node, not every node), which means you need to pass `this` to `getHeadingOrder` or use an arrow function instead of a named function. That complicates things and only adds further nesting, so I think we should revert and leave this case as is.